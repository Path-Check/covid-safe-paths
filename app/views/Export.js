import React, { Component } from 'react';
import {
    SafeAreaView,
    StyleSheet,
    ScrollView,
    Linking,
    View,
    Text,
    Alert,
    Image
} from 'react-native';

import colors from "../constants/colors";
import WebView from 'react-native-webview';
import Button from "../components/Button";
import { GetStoreData } from '../helpers/General';
import Share from 'react-native-share';
import RNFetchBlob from 'rn-fetch-blob';

const base64 = RNFetchBlob.base64


//import RNShareFile from 'react-native-file-share';


class ExportScreen extends Component {
    constructor(props) {
        super(props);
    }

    onShare = async () => {
            try {
                const locationArray = await GetStoreData('LOCATION_DATA');
                var locationData;

                if (locationArray !== null) {
                    locationData = JSON.parse(locationArray);
                } else {
                    locationData = [];
                }

                b64Data = base64.encode(JSON.stringify(locationData));
                Share.open({
                    url: "data:string/txt;base64," + b64Data
                }).then(res => {
                    console.log(res);
                })
                .catch(err => {
                    console.log(err.message, err.code);
                })
            } catch (error) {
                console.log(error.message);
        }
    };

    backToMain() {
        this.props.navigation.navigate('LocationTrackingScreen', {})
    }

    render() {

        return (
            <SafeAreaView style={styles.container}>
                <View style={styles.headerContainer}>
                    <Text onPress={() => this.backToMain()} style={styles.backArrow}> &#8249; </Text>
                    <Text style={styles.sectionDescription}>Export</Text>
                </View>


                <View style={styles.main}>
                    <Text style={styles.mainText}>You can share you location trail with anyone using the Share button below.  Once you press the button it will ask you with whom and how you want to share it.</Text>
                    <Text style={styles.mainText}>Location is shared as a simple list of times and coordinates, no other identifying information.</Text>
                    <View style={styles.block}>
                        <Button onPress={this.onShare} title="Share" />
                    </View>
                </View>

            </SafeAreaView>
        )
    }
}

const styles = StyleSheet.create({
    // Container covers the entire screen
    container: {
        flex: 1,
        flexDirection: 'column',
        color: colors.PRIMARY_TEXT,
        backgroundColor: colors.APP_BACKGROUND,
    },
    headerTitle: {
        textAlign: 'center',
        fontWeight: "bold",
        fontSize: 38,

        padding: 0
    },
    subHeaderTitle: {
        textAlign: 'center',
        fontWeight: "bold",
        fontSize: 22,
        padding: 5
    },
    main: {
        flex: 1,
        flexDirection: 'column',
        textAlignVertical: 'top',
        alignItems: 'center',
        width: "95%"
    },
    block: {
        margin: 20,
        width: "75%",
        alignItems: 'center',
        justifyContent: 'center'
    },
    mainText: {
        fontSize: 18,
        lineHeight: 24,
        fontWeight: '400',
        textAlignVertical: 'center',
        padding: 20,
    },

    headerContainer: {
        flexDirection: 'row',
    },
    backArrow: {
        fontSize: 60,
        lineHeight: 60,
        fontWeight: '400',
        marginRight: 5,
        textAlignVertical: 'center'
    },
    sectionDescription: {
        fontSize: 24,
        lineHeight: 24,
        fontWeight: '800',
        textAlignVertical: 'center'
    },
});

export default ExportScreen;
