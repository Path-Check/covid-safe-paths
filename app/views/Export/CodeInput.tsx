import React, { useState, MutableRefObject, useRef, useEffect } from 'react';
import {
  TextInput,
  NativeSyntheticEvent,
  TextInputKeyPressEventData,
  View,
  StyleSheet,
} from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';

export const CodeInput = ({
  code,
  length,
  setCode,
}: {
  code: string;
  length: number;
  setCode: (code: string) => void;
}): JSX.Element => {
  const [currentIndex, setCurrentIndex] = useState(0);

  const characters: string[] = [];
  for (let i = 0; i < length; i++) characters.push(code[i]);

  const characterRefs: MutableRefObject<TextInput[]> = useRef([]);
  useEffect(() => {
    characterRefs.current = characterRefs.current.slice(0, length);
  }, [length]);

  const focus = (i: number) => {
    characterRefs.current[i].focus();
  };

  // Focus on mount
  useEffect(() => {
    setTimeout(() => {
      focus(0);
    }, 0); // allow waiting for transition to end & first paint
  }, []);

  const onFocus = (i: number) => {
    if (i > currentIndex) {
      // prohibit skipping forward
      focus(currentIndex);
    } else {
      // restart at clicked character
      setCurrentIndex(i);
      setCode(code.slice(0, i));
    }
  };

  // Adding characters
  const onChangeCharacter = (d: string) => {
    if (d.length) {
      setCode(code.slice(0, currentIndex) + d);
      const nextIndex = currentIndex + 1;
      if (nextIndex < length) {
        setCurrentIndex(nextIndex);
        focus(nextIndex);
      }
    }
  };

  // Removing characters
  const onKeyPress = (e: NativeSyntheticEvent<TextInputKeyPressEventData>) => {
    if (e.nativeEvent.key === 'Backspace') {
      // go to previous
      if (!code[currentIndex]) {
        const newIndex = currentIndex - 1;
        if (newIndex >= 0) {
          setCurrentIndex(newIndex);
          setCode(code.slice(0, newIndex));
          focus(newIndex);
        }
      }
      // clear current (used for last character)
      else {
        setCode(code.slice(0, currentIndex));
      }
    }
  };

  return (
    <View style={{ flexDirection: 'row', flexShrink: 1 }}>
      {characters.map((character, i) => (
        <TextInput
          ref={(ref) => {
            if (ref != null) {
              characterRefs.current[i] = ref;
            }
          }}
          key={`${i}CodeCharacter`}
          value={character}
          style={[
            styles.characterInput,
            {
              borderColor: character
                ? Colors.primaryBorder
                : Colors.quaternaryViolet,
            },
          ]}
          keyboardType={'number-pad'}
          returnKeyType={'done'}
          onChangeText={onChangeCharacter}
          onKeyPress={onKeyPress}
          blurOnSubmit={false}
          onFocus={() => onFocus(i)}
          testID={`input${i}`}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  characterInput: {
    fontSize: 20,
    color: Colors.violetTextDark,
    textAlign: 'center',
    flexGrow: 1,
    aspectRatio: 1,
    borderWidth: 2,
    borderRadius: 10,
    marginRight: 6,
  },
});

export default CodeInput;
