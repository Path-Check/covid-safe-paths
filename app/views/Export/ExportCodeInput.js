// TODO: REMOVE
/* eslint-disable react-native/no-raw-text */
import React, { useEffect, useRef, useState } from 'react';
import {
  KeyboardAvoidingView,
  StatusBar,
  StyleSheet,
  TextInput,
  View,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import { Icons } from '../../assets';
import { Button } from '../../components/Button';
import { IconButton } from '../../components/IconButton';
import { Typography } from '../../components/Typography';
import Colors from '../../constants/colors';
import fontFamily from '../../constants/fonts';
import { Theme } from '../../constants/themes';

const CODE_LENGTH = 6;

const CodeInput = ({ code, length, setCode }) => {
  const [currentIndex, setCurrentIndex] = useState(0);

  let digits = [];
  for (let i = 0; i < length; i++) digits.push(code[i]);

  const digitRefs = useRef([]);
  useEffect(() => {
    digitRefs.current = digitRefs.current.slice(0, length);
  }, [length]);

  const focus = i => {
    digitRefs.current[i].focus();
  };

  // Focus on mount
  useEffect(() => {
    setTimeout(() => {
      focus(0);
    }, 0); // allow waiting for transition to end & first paint
  }, []);

  const onFocus = i => {
    if (i > currentIndex) {
      // prohibit skipping forward
      focus(currentIndex);
    } else {
      // restart at clicked digit
      setCurrentIndex(i);
      setCode(code.slice(0, i));
    }
  };

  // Adding digits
  const onChangeDigit = d => {
    if (d.length) {
      setCode(code.slice(0, currentIndex) + d);
      const nextIndex = currentIndex + 1;
      if (nextIndex < length) {
        setCurrentIndex(nextIndex);
        focus(nextIndex);
      }
    }
  };

  // Removing digits
  const onKeyPress = e => {
    if (e.nativeEvent.key === 'Backspace') {
      // go to previous
      if (!code[currentIndex]) {
        const newIndex = currentIndex - 1;
        if (newIndex >= 0) {
          setCurrentIndex(newIndex);
          setCode(code.slice(0, newIndex));
          focus(newIndex);
        }
      }
      // clear current (used for last digitf)
      else {
        setCode(code.slice(0, currentIndex));
      }
    }
  };

  return (
    <View style={{ flexDirection: 'row', flex: 1 }}>
      {[...digits].map((digit, i) => (
        <TextInput
          ref={ref => (digitRefs.current[i] = ref)}
          key={`${i}CodeDigit`}
          value={digit}
          style={{
            fontSize: 20,
            color: '#1F2C9B',
            textAlign: 'center',
            flexShrink: 1,
            aspectRatio: 1,
            borderWidth: 2,
            borderColor: digit ? Colors.VIOLET_BUTTON : '#E5E7FA',
            borderRadius: 10,
            marginRight: 6,
          }}
          keyboardType={'number-pad'}
          returnKeyType={'done'}
          onChangeText={onChangeDigit}
          onKeyPress={onKeyPress}
          blurOnSubmit={false}
          onFocus={() => onFocus(i)}
        />
      ))}
    </View>
  );
};

export const ExportSelectHA = ({ route, navigation }) => {
  const [code, setCode] = useState('');
  const { selectedAuthority } = route.params;

  console.log({ code });

  return (
    <Theme use='default'>
      <StatusBar
        barStyle='dark-content'
        backgroundColor={Colors.VIOLET_BUTTON}
        translucent={false}
      />
      <SafeAreaView
        style={{ flex: 1, paddingBottom: 44, backgroundColor: '#F8F8FF' }}>
        <KeyboardAvoidingView
          behavior={'padding'}
          style={{ paddingHorizontal: 24, paddingBottom: 20, flex: 1 }}>
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'space-between',
              padding: 12,
              paddingLeft: 0,
            }}>
            <IconButton
              icon={Icons.BackArrow}
              size={27}
              onPress={() => navigation.replace('ExportSelectHA')}
            />
            <IconButton
              icon={Icons.Close}
              size={22}
              onPress={() => navigation.navigate('SettingsScreen')}
            />
          </View>
          <View style={{ flex: 1 }}>
            <Typography use='headline2' style={styles.exportSectionTitles}>
              Enter your verification code
            </Typography>
            <View style={{ height: 8 }} />
            <Typography use='body1'>
              {`The representative from ${selectedAuthority.name} will provide a verfication code over the phone to link your data with ${selectedAuthority.name}.`}
            </Typography>
            <View style={{ height: 60 }} />
            <CodeInput code={code} length={CODE_LENGTH} setCode={setCode} />
            <Button
              style={{ marginBottom: 20 }}
              disabled={code.length < CODE_LENGTH}
              label={'Next'}
              onPress={() => {}}
            />
          </View>
        </KeyboardAvoidingView>
      </SafeAreaView>
    </Theme>
  );
};

const styles = StyleSheet.create({
  exportSectionTitles: {
    color: '#1F2C9B',
    fontWeight: '500',
    fontFamily: fontFamily.primaryMedium,
  },
});

export default ExportSelectHA;
