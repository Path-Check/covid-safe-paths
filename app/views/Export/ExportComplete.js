import React from 'react';
import { useTranslation } from 'react-i18next';

import { useAssets } from '../../TracingStrategyAssets';
import ExportTemplate from './ExportTemplate';
import { isGPS } from '../../COVIDSafePathsConfig';
import { Screens } from '../../navigation';

export const ExportComplete = ({ navigation }) => {
  const { t } = useTranslation();
  const { exportCompleteBody } = useAssets();
  const onClose = () => navigation.navigate(exportExitRoute);

  const exportExitRoute = isGPS ? Screens.ExportStart : Screens.Settings;

  return (
    <ExportTemplate
      lightTheme
      onNext={onClose}
      nextButtonLabel={t('common.done')}
      headline={t('export.complete_title')}
      body={exportCompleteBody}
    />
  );
};

export default ExportComplete;
