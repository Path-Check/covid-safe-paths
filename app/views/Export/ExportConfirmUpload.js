import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Alert, NativeModules } from 'react-native';

import { Icons } from '../../assets';
import exitWarningAlert from './exitWarningAlert';
import ExportTemplate from './ExportTemplate';

const MOCK_ENDPOINT =
  'https://private-anon-da01e87e46-safeplaces.apiary-mock.com/upload';

export const ExportComplete = ({ navigation, route }) => {
  const { t } = useTranslation();
  const [isUploading, setIsUploading] = useState(false);
  const onClose = () => exitWarningAlert(navigation);

  const { selectedAuthority, code } = route.params;

  const upload = async () => {
    setIsUploading(true);
    try {
      const concernPoints = await NativeModules.SecureStorageManager.getLocations();
      const res = await fetch(MOCK_ENDPOINT, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ accessCode: code, concernPoints }),
      });
      const success = res.status === 201;
      if (success) {
        navigation.navigate('ExportComplete');
      } else {
        throw res.status;
      }
      setIsUploading(false);
    } catch (e) {
      Alert.alert(t('common.something_went_wrong'));
      setIsUploading(false);
    }
  };

  return (
    <ExportTemplate
      onClose={onClose}
      onNext={upload}
      nextButtonLabel={t('export.confirm_upload_button')}
      headline={t('export.confirm_upload_title')}
      body={t('export.confirm_upload_body', { name: selectedAuthority.name })}
      icon={Icons.Upload}
      buttonLoading={isUploading}
    />
  );
};

export default ExportComplete;
