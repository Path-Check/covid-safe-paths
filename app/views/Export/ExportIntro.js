import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';

import getHealthcareAuthorities from '../../store/actions/healthcareAuthorities/getHealthcareAuthoritiesAction';
import healthcareAuthorityOptionsSelector from '../../store/selectors/healthcareAuthorityOptionsSelector';
import { useAssets } from '../../TracingStrategyAssets';
import ExportTemplate from './ExportTemplate';
import { Icons } from '../../assets';

export const ExportIntro = ({ navigation }) => {
  const { t } = useTranslation();
  const { exportStartTitle, exportStartBody } = useAssets();

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getHealthcareAuthorities());
  }, [dispatch]);
  const authorities = useSelector(healthcareAuthorityOptionsSelector);
  const selectedAuthorityDummy = authorities[0];

  const onNext = () =>
    navigation.navigate('ExportCodeInput', {
      selectedAuthority: selectedAuthorityDummy,
    });
  const onClose = () => navigation.navigate('SettingsScreen');
  return (
    <ExportTemplate
      onNext={onNext}
      onClose={onClose}
      icon={Icons.Heart}
      headline={exportStartTitle}
      body={exportStartBody}
      nextButtonLabel={t('common.start')}
      ignoreModalStyling // this is in a tab
    />
  );
};

export default ExportIntro;
