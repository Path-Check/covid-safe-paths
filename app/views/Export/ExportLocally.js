import PropTypes from 'prop-types';
import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  NativeModules,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  View,
} from 'react-native';
import RNFS from 'react-native-fs';
import LinearGradient from 'react-native-linear-gradient';
import Share from 'react-native-share';
import RNFetchBlob from 'rn-fetch-blob';

import { Button, Typography } from '../../components';
import fontFamily from '../../constants/fonts';
import { Theme } from '../../constants/themes';
import { isPlatformiOS } from '../../Util';

import { Icons } from '../../assets';
import { Colors } from '../../styles';

const base64 = RNFetchBlob.base64;

// NOTE:
// This is the old way we export. This is still the default, but will
// become flipped behind the feature flag once we have staging for uploading.
const ExportLocally = () => {
  const { t } = useTranslation();

  async function onShare() {
    try {
      let locationData = await NativeModules.SecureStorageManager.getLocations();
      let nowUTC = new Date().toISOString();
      let unixtimeUTC = Date.parse(nowUTC);

      let options = {};
      let jsonData = JSON.stringify(
        locationData.map(({ latitude, longitude, time }) => ({
          latitude,
          longitude,
          time,
        })),
      );
      const title = 'COVIDSafePaths.json';
      const filename = unixtimeUTC + '.json';
      const message = 'Here is my location log from PathCheck.';
      if (isPlatformiOS()) {
        const url = RNFS.DocumentDirectoryPath + '/' + filename;
        await RNFS.writeFile(url, jsonData, 'utf8')
          .then(() => {
            options = {
              activityItemSources: [
                {
                  placeholderItem: { type: 'url', content: url },
                  item: {
                    default: { type: 'url', content: url },
                  },
                  subject: {
                    default: title,
                  },
                  linkMetadata: { originalUrl: url, url, title },
                },
              ],
            };
          })
          .catch((err) => {
            console.log(err.message);
          });
      } else {
        jsonData = 'data:application/json;base64,' + base64.encode(jsonData);
        options = {
          title,
          subject: title,
          url: jsonData,
          message: message,
          filename: filename,
        };
      }
      await Share.open(options)
        .then((res) => {
          console.log(res);
        })
        .catch((err) => {
          console.log(err);
          console.log(err.message, err.code);
        });
      if (isPlatformiOS()) {
        // eslint-disable-next-line no-undef
        await RNFS.unlink(url);
      }
    } catch (error) {
      console.log(error.message);
    }
  }

  return (
    <Theme use='violet'>
      <StatusBar
        barStyle='light-content'
        backgroundColor={Colors.secondaryBlue}
        translucent={false}
      />
      <SafeAreaView style={styles.topSafeAreaContainer} />
      <SafeAreaView style={styles.bottomSafeAreaContainer}>
        <LinearGradient
          start={{ x: 0, y: 0 }}
          end={{ x: 0, y: 1 }}
          colors={[Colors.secondaryBlue, Colors.primaryBlue]}
          style={{ flex: 1, height: '100%' }}>
          <ScrollView contentContainerStyle={styles.contentContainer}>
            <View style={styles.main}>
              <Typography use='headline2' style={styles.exportSectionTitles}>
                {t('share.title')}
              </Typography>
              <Typography use='body1' style={styles.exportSectionPara}>
                {t('share.paragraph_first')}
              </Typography>
              <Typography use='body1' style={styles.exportSectionPara}>
                {t('share.paragraph_second')}
              </Typography>

              <Button
                style={styles.exportButton}
                label={t('share.button_text')}
                icon={Icons.Export}
                onPress={onShare}
              />
            </View>
          </ScrollView>
        </LinearGradient>
      </SafeAreaView>
    </Theme>
  );
};

const styles = StyleSheet.create({
  // Container covers the entire screen
  topSafeAreaContainer: {
    flex: 0,
    backgroundColor: Colors.secondaryBlue,
  },
  bottomSafeAreaContainer: {
    flex: 1,
    backgroundColor: Colors.primaryBlue,
  },
  contentContainer: {
    flexDirection: 'column',
    width: '100%',
    flex: 1,
    paddingHorizontal: 26,
  },
  exportSectionTitles: {
    marginTop: 9,
    fontWeight: 'normal',
    fontFamily: fontFamily.primaryMedium,
  },
  exportSectionPara: {
    marginTop: 22,
  },
  exportButton: {
    marginTop: 48,
  },
  main: {
    flex: 1,
    paddingTop: 48,
  },
});

ExportLocally.propTypes = {
  shareButtonDisabled: PropTypes.bool,
};

ExportLocally.defaultProps = {
  shareButtonDisabled: true,
};

export default ExportLocally;
