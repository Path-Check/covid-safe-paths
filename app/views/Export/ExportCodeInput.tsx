import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import {
  Alert,
  KeyboardAvoidingView,
  StatusBar,
  StyleSheet,
  View,
  Platform,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import { Button } from '../../components/Button';
import { IconButton } from '../../components/IconButton';
import { Typography } from '../../components/Typography';
import { Theme } from '../../constants/themes';
import exitWarningAlert from './exitWarningAlert';
import exportCodeApi from '../../api/export/exportCodeApi';
import { Screens } from '../../navigation';
import { isGPS } from '../../COVIDSafePathsConfig';
import { useStrategyContent } from '../../TracingStrategyContext';

import { Icons } from '../../assets';
import { Colors } from '../../styles';
import {
  NavigationScreenProp,
  NavigationRoute,
  NavigationState,
  NavigationParams,
} from 'react-navigation';
import { HealthcareAuthority } from '../../store/types';
import CodeInput from './CodeInput';

const CODE_LENGTH = 6;

export const ExportCodeInput = ({
  route,
  navigation,
}: {
  navigation: NavigationScreenProp<NavigationState, NavigationParams>;
  route: NavigationRoute;
}): JSX.Element => {
  const { t } = useTranslation();
  const { InterpolatedStrategyCopy, StrategyCopy } = useStrategyContent();

  const exportCodeInputNextRoute = isGPS
    ? Screens.ExportLocationConsent
    : Screens.PublishConsent;

  const exportExitRoute = isGPS ? Screens.ExportStart : Screens.Settings;

  const [code, setCode] = useState('');
  const [isCheckingCode, setIsCheckingCode] = useState(false);
  const [codeInvalid, setCodeInvalid] = useState(false);

  const selectedAuthority: HealthcareAuthority =
    route.params?.selectedAuthority;
  const validateCode = async () => {
    setIsCheckingCode(true);
    setCodeInvalid(false);
    try {
      if (isGPS) {
        const { valid } = await exportCodeApi(selectedAuthority, Number(code));

        if (valid) {
          navigation.navigate(exportCodeInputNextRoute, {
            selectedAuthority,
            code,
          });
        } else {
          setCodeInvalid(true);
        }
        setIsCheckingCode(false);
      } else {
        const valid = code === '123456';

        if (valid) {
          navigation.navigate(exportCodeInputNextRoute, {
            selectedAuthority,
            code,
          });
        } else {
          setCodeInvalid(true);
        }
        setIsCheckingCode(false);
      }
    } catch (e) {
      Alert.alert(t('common.something_went_wrong'), e.message);
      setIsCheckingCode(false);
    }
  };

  return (
    <Theme use='default'>
      <StatusBar
        barStyle='dark-content'
        backgroundColor={Colors.primaryBackgroundFaintShade}
        translucent={false}
      />
      <SafeAreaView style={styles.wrapper}>
        <KeyboardAvoidingView behavior={'padding'} style={styles.container}>
          <View style={styles.headerIcons}>
            <IconButton
              icon={Icons.BackArrow}
              size={27}
              onPress={() => navigation.goBack()}
            />
            <IconButton
              icon={Icons.Close}
              size={22}
              onPress={() => exitWarningAlert(navigation, exportExitRoute)}
            />
          </View>
          <View style={{ flex: 1, marginBottom: 20 }}>
            <Typography use='headline2'>
              {StrategyCopy.exportCodeTitle}
            </Typography>
            <View style={{ height: 8 }} />
            <Typography use='body1'>
              {InterpolatedStrategyCopy.exportCodeBody(selectedAuthority.name)}
            </Typography>
            {/* These flex grows allow for a lot of flexibility across device sizes */}
            <View style={{ maxHeight: 60, flexGrow: 1 }} />
            {/* there's a flex end bug on android, this is a hack to ensure some spacing */}
            <View
              style={{
                flexGrow: 1,
                marginVertical: Platform.OS === 'ios' ? 0 : 10,
              }}>
              <CodeInput code={code} length={CODE_LENGTH} setCode={setCode} />
              {codeInvalid && (
                <Typography style={styles.errorSubtitle} use='body2'>
                  {t('export.code_input_error')}
                </Typography>
              )}
            </View>
            <Button
              disabled={code.length < CODE_LENGTH}
              loading={isCheckingCode}
              label={t('common.next')}
              onPress={validateCode}
              testID='next-button'
            />
          </View>
        </KeyboardAvoidingView>
      </SafeAreaView>
    </Theme>
  );
};

const styles = StyleSheet.create({
  errorSubtitle: {
    marginTop: 8,
    color: Colors.errorText,
  },
  headerIcons: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 12,
    paddingLeft: 0,
  },
  wrapper: {
    flex: 1,
    paddingBottom: 44,
    backgroundColor: Colors.primaryBackgroundFaintShade,
  },
  container: {
    paddingHorizontal: 24,
    paddingBottom: 20,
    flex: 1,
  },
});

export default ExportCodeInput;
