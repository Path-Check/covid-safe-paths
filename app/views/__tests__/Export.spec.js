import { render } from '@testing-library/react-native';
import React from 'react';

import { ExportScreen } from '../Export';

describe('<Export />', () => {
  it('renders correctly', () => {
    const { asJSON } = render(<ExportScreen />);
    expect(asJSON()).toMatchSnapshot();
  });
});
