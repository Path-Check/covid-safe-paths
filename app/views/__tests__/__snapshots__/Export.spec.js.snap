// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportScreen /> renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#6979f8",
        "flex": 0,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#4051db",
        "flex": 1,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4285102584,
          4282405339,
        ]
      }
      endPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "flex": 1,
          "height": "100%",
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "paddingHorizontal": 26,
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flex": 1,
                "paddingTop": 48,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "IBMPlexSans-Medium",
                  "fontSize": 26,
                  "fontWeight": "normal",
                  "lineHeight": 34,
                  "marginTop": 9,
                  "writingDirection": "ltr",
                }
              }
              use="headline2"
            >
              Share location history
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "IBMPlexSans",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 24,
                  "marginTop": 22,
                  "writingDirection": "ltr",
                }
              }
              use="body1"
            >
              If you test positive for COVID-19, please do your part by sharing your location history with local authorities.
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "IBMPlexSans",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 24,
                  "marginTop": 22,
                  "writingDirection": "ltr",
                }
              }
              use="body1"
            >
              Location is shared as a simple list of times and places, no additional information.
            </Text>
            <View
              accessibilityLabel="Share location data"
              accessibilityRole="button"
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#ffffff",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderColor": "transparent",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "height": 72,
                  "justifyContent": "space-between",
                  "marginTop": 48,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#4051db",
                    "fontFamily": "IBMPlexSans-Medium",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "lineHeight": 40,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Share location data
              </Text>
              <SvgXml
                style={
                  Object {
                    "marginRight": 8,
                  }
                }
                xml="<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<svg width=\\"16\\" height=\\"21\\" viewBox=\\"0 0 16 21\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<path d=\\"M13 7H14C15.1046 7 16 7.89543 16 9V19C16 20.1046 15.1046 21 14 21H2C0.89543 21 0 20.1046 0 19V9C0 7.89543 0.89543 7 2 7H3C3.55228 7 4 7.44772 4 8C4 8.55228 3.55228 9 3 9H2V19H14V9H13C12.4477 9 12 8.55228 12 8C12 7.44772 12.4477 7 13 7ZM7.24029 0.349711C7.42369 0.135645 7.69601 0 8 0C8.30399 0 8.57631 0.135645 8.75971 0.349711L11.5355 3.12553C11.9261 3.51606 11.9261 4.14922 11.5355 4.53975C11.145 4.93027 10.5118 4.93027 10.1213 4.53975L9 3.41843V13C9 13.5523 8.55228 14 8 14C7.44772 14 7 13.5523 7 13V3.41843L5.87868 4.53975C5.48815 4.93027 4.85499 4.93027 4.46447 4.53975C4.07394 4.14922 4.07394 3.51606 4.46447 3.12553L7.24029 0.349711Z\\" fill=\\"#4051DB\\"/>
</svg>"
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </BVLinearGradient>
  </RCTSafeAreaView>
</View>
`;
